contentcuration/contentcuration/frontend/channelEdit/router.js

```diff
@@ -85,6 +87,18 @@ const router = new VueRouter({
           props: true,
           component: EditModal,
         },
+        {
```

Comment:
In order to preserve the current mode, could you copy/paste this a children for all the EditModal routes? Something like
```
{
    name: RouterNames.CONTENTNODE_DETAILS,
    ...
    children: [related routes]
},
{
    name: RouterNames.UPLOAD_FILES,
    ... 
    children: [related routes]
```

Then on the cancel button, you can use something like
```
previousLink() {
    return {
        name: this.$route.matched[this.$route.matched.length - 2].name,
        params: {
           ...this.$route.params,
           targetNodeId: undefined
       }
}
```

I know this isn't the cleanest solution by any stretch, but I would like to prioritize preserving the current mode and iterating on it later (once we migrate the edit modal, we can also look into making this cleaner)

Reply:
No worries :)

I think I could do that though it seems that I will need to replace `targetNodeId` by `detailNodeIds` in [AddRelatedResourcesModal](https://github.com/learningequality/studio/pull/1787/files#diff-bef8643dc4d5e631a675e73b33593943R108), [AddPreviousStepsModal](https://github.com/learningequality/studio/pull/1787/files#diff-bef8643dc4d5e631a675e73b33593943R108) and [AddNextStepsModal](https://github.com/learningequality/studio/pull/1787/files#diff-8460b7bc860c01bcc4a088c632d7f642R25) and get first node from there. It should not be technically difficult but it will again add more components with interface seemingly working with more node IDs while designed to work only with one from both UI and technical perspective. Not sure if we want to add more of these? Any ideas on how to solve it in a reasonable way that would allow them to work with one node ID? I will continue with trying to figure it out tomorrow and if I have no idea, I will just switch the interface to accept more ids, just wanted to point it out and ask if someone has any ideas before I get back to that tomorrow.

Reply:
@jayoshih

I probably misunderstood the comment (though notes above are still valid). Could you please clarify in more detail the proposed behavior? We don't have `<router-view>` inside of `EditModal` so I am not sure how this router structure with children could work there? Or do you want to add `<router-view>` there? I am afraid that would complicate things even more as EditModal is rendered within `TreeView`'s `<router-view>` and there's some custom routing logic in router hooks in `EditModal`.

Another thing is that even if we're somehow able to achieve that, it would require placing two fullscreen dialog components on top of each other which we wanted to avoid. I am also not sure how that would work when we migrate to the immersive modal layouts.

If I've missed something and there is a solution to this, even a bit hackish, I don't mind adding it but right now I can't see any other reasonable way than to update this behavior as we move `EditModal` to `/frontend`. It doesn't need to necessarily be a huge refactor as we discussed, just something that would solve the issues mentioned above. I can have a look at it in a follow-up too.

If you have an idea or see how to easily address these things at this point, might be better to plan a call one more time so let me know anytime :)

Reply:
FYI @jayoshih and I agreed to solve this as follow-up so resolving.

Reply:
Follow-up issue https://github.com/learningequality/studio/issues/1816